app/src/main/java/fr/free/nrw/commons/contributions/ContributionsLocalDataSource.java

```diff
@@ -1,49 +1,80 @@
 package fr.free.nrw.commons.contributions;
 
-import android.database.Cursor;
+import android.app.job.JobScheduler;
+import android.text.TextUtils;
+
+import androidx.lifecycle.LiveData;
+
+import java.util.List;
 
 import javax.inject.Inject;
 import javax.inject.Named;
 
+import fr.free.nrw.commons.db.AppDatabase;
 import fr.free.nrw.commons.kvstore.JsonKvStore;
+import io.reactivex.Single;
 
 /**
  * The LocalDataSource class for Contributions
  */
 class ContributionsLocalDataSource {
 
-    private final ContributionDao contributionsDao;
+    private final ContributionDao contributionDao;
     private final JsonKvStore defaultKVStore;
 
     @Inject
     public ContributionsLocalDataSource(
             @Named("default_preferences") JsonKvStore defaultKVStore,
-            ContributionDao contributionDao) {
```

Comment:
Just inject ContributionDao, testing will be a little bit more annoying if there is some indirection with the object's dependencies